compiler/src/dotty/tools/dotc/transform/init/Objects.scala

```diff
@@ -392,7 +447,11 @@ object Objects:
         // TODO: handle RefSet
         thisV match
         case ref: OfClass =>
-          resolveEnv(meth, ref.outer, ref.env)
+          ref.outer match
+          case outer : ThisValue =>
+            resolveEnv(meth, outer, ref.env)
+          case _ => // This is the case for top-level
+            None
```

Comment:
What is the type of `ref.outer`, may we change it to a more specific type to avoid the pattern match here?

Reply:
Based on your email, I think the possible types for `ref.outer` is `ThisValue` or `Bottom`, and the `Bottom` case is only for top-level classes. I wrote this pattern-match so that the second parameter can be narrowed down as `ThisValue`, and we stop searching when reaching a top-level class as before